File: app/src/main/java/com/example/note/ui/fragments/EditableFragment.java
package com.example.note.ui.fragments;

import android.os.Bundle;
import android.text.Editable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.note.R;
import com.example.note.viewmodel.NotesListViewModel;
import com.google.android.material.textfield.TextInputEditText;

public class EditableFragment extends Fragment {
    NotesListViewModel viewModel;
    EditText editNoteDescription;
    TextInputEditText editNoteTitle;

    @Nullable
    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_editable, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        viewModel = new ViewModelProvider(requireActivity()).get(NotesListViewModel.class);
        editNoteTitle = view.findViewById(R.id.edit_note_title);
        editNoteDescription = view.findViewById(R.id.edit_note_description);

        editNoteTitle.addTextChangedListener(new TextWatcher());
        editNoteDescription.addTextChangedListener(new TextWatcher());

        viewModel.getSelectedNote().observe(getViewLifecycleOwner(), note -> {
            editNoteTitle.setText((note == null) ? "" : note.getTitle());
            editNoteDescription.setText((note == null) ? "" : note.getDescription());
        });
    }

    @Override
    public void onDetach() {
        super.onDetach();
        viewModel.noteClose();
    }

    static class TextWatcher implements android.text.TextWatcher {
        @Override
        public void beforeTextChanged(CharSequence s, int start, int count, int after) {

        }

        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {

        }

        @Override
        public void afterTextChanged(Editable s) {
            //TODO --- save text
        }
    }
}

File: app/src/main/java/com/example/note/ui/fragments/EditableFragment.kt
package com.example.note.ui.fragments

import android.os.Bundle
import android.text.Editable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.note.R
import com.example.note.viewmodel.NotesListViewModel
import com.google.android.material.textfield.TextInputEditText

class EditableFragment : Fragment() {
    private lateinit var viewModel: NotesListViewModel
    private var editNoteDescription: EditText? = null
    private var editNoteTitle: TextInputEditText? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_editable, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel = ViewModelProvider(requireActivity()).get(NotesListViewModel::class.java)
        editNoteTitle = view.findViewById(R.id.edit_note_title)
        editNoteDescription = view.findViewById(R.id.edit_note_description)

        editNoteTitle?.addTextChangedListener(TextWatcher())
        editNoteDescription?.addTextChangedListener(TextWatcher())

        viewModel.getSelectedNote().observe(viewLifecycleOwner) { note ->
            editNoteTitle?.setText(note.title)
            editNoteDescription?.setText(note.description)
        }
    }

    override fun onDetach() {
        super.onDetach()
        viewModel.noteClose()
    }

    class TextWatcher : android.text.TextWatcher {
        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {

        }

        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {

        }

        override fun afterTextChanged(s: Editable?) {
            //TODO(save text)
        }
    }
}